packages/wallet-adapter-react/src/WalletProvider.tsx

```diff
@@ -45,14 +45,19 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
   const [{ connected, account, network, wallet }, setState] =
     useState(initialState);
 
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
   const walletCore = useMemo(() => new WalletCore(plugins), []);
   const [wallets, setWallets] = useState<Wallet[]>(walletCore.wallets);
 
   const connect = (walletName: WalletName) => {
     try {
+      setIsLoading(true);
       walletCore.connect(walletName);
     } catch (e) {
       console.log("connect error", e);
```

Comment:
Probably not here but in a separate PR, could we also expose the error? Something like:
```
const [error, setError] = useState<SomeStructuredErrorType | undefined>(undefined);
```